kratos/python/add_utilities_to_python.cpp

```diff
@@ -524,13 +524,31 @@ void AddUtilitiesToPython(pybind11::module& m)
         .def("UpdateSearchDatabase", &BinBasedFastPointLocator < 2 > ::UpdateSearchDatabase)
         .def("UpdateSearchDatabaseAssignedSize", &BinBasedFastPointLocator < 2 > ::UpdateSearchDatabaseAssignedSize)
         .def("FindPointOnMesh", &BinBasedFastPointLocator < 2 > ::FindPointOnMeshSimplified)
+        .def("FindPointOnMesh", [](BinBasedFastPointLocator < 2 >& self, const array_1d<double,3>& coords, const double tol)
+            {
+                Element::Pointer pelem;
+                Vector N(3);
```

Comment:
You are assuming this is a triangle, @bodhinandach and I updated the interface to allow consider any geometry, like quads

Reply:
+1 for @loumalouomega 

Also the name should indicate that you are searching for elements, I think recently there was an opportunity added to use it also for conditions 
Right @adityaghantasala ?

Reply:
Yep

Reply:
well, i guess that you do a resize of N internally correct? if that is so, would it be enough to just define

      Vector N;

without size?

Reply:
about the version for conditions, i don't think it can be this same function...

Reply:
Yes it is possible to define Vector N without size @RiccardoRossi

Reply:
ok, i changed the size of N.

about the version for conditions, i don't think it can be this same function...

Reply:
> about the version for conditions, i don't think it can be this same function...

No, ofc not, but imo the name of the fct should contain „Element“

Reply:
I agree that name should contain "Element" .. !

Reply:
A general question : `max_results = 1000 `
is this valid for all the cases ?

Reply:
@philbucher we cannot change the name easily (I mean as a part of this PR), since the utility is used through the code... 

Not saying that i oppose to the change, just that it is a much more complex PR than the one at hand. I think that we should merge this one (which only provides a simpler user interface FROM PYTHON) and maybe you could open an issue as reminder that the change you propose would be due

Reply:
I meant only for the function that you are adding

And eventually the other fcts could also be updated, but as part of a different PR ofc

Reply:
@adityaghantasala this is a simplified function, so i used what i considered a reasonable (actually quite large) buffer. I will create two versions, one with tol and nresults and one without additional parameters

@philbucher for the name, it is not the name of the function to be changed, but rather the name of the object...